### ISSUE TITLE
More specific class names for default collectors

### ISSUE THREAD

**Issue by BigRoy** [pyblish/pyblish-maya#52]
Not entirely sure whether this could be a problem but the default plug-ins in Pyblish Maya use pretty generic class names `SelectCurrentFile` and `SelectWorkspace` (or `CollectCurrentFile` and `CollectWorkspace` in #51).

If I'm correct Pyblish uses the class names to identify "duplicate plug-ins" and as such this could quickly conflict with plug-ins that others might develop themselves. In the end we could assume they'd just "not use" the default plug-ins or we can target the default plug-in class names a bit more towards Maya.

For example `CollectMayaCurrentFile` and `CollectMayaWorkspace`.

What's the intended workflow regarding default plug-ins in integrations?

**Comment by mottosso** [comment]
Well spotted; I think you're right. Could you confirm whether it actually does discard duplicates in this way?

**Comment by BigRoy** [comment]
> Could you confirm whether it actually does discard duplicates in this way?

So I made a new plug-in with a different file name and basically everything different (docstrings, version, etc.) but only kept the class name the same.

```python
import pyblish.api


@pyblish.api.log
class CollectCurrentFile(pyblish.api.ContextPlugin):

    hosts = ['maya']
    version = (1, 1, 0)

    def process(self, context):
        print "TEST"
```

ended up getting a duplicate plug-in found message:

```
# pyblish.plugin : Duplicate plug-in found: <class 'collect_current_file.CollectCurrentFile'> # 
```

So I'm assuming it is. 

I'm not sure if the `version` on the plug-ins is actually being used or is just a leftover from the beginning of Pyblish. Because increasing the version number on my plug-in compared to the one in Pyblish Maya or the other way around didn't make the other to take precedence. In my particular case my custom plug-in took it. Lucky? I think who's the winner is based on order of loading plug-ins; I'm not sure.

**Comment by tokejepsen** [comment]
Do you mean it would conflict with other hosts default plugins?

Surely people would just use the data collected by those plugins, or they would need the plugin hence not having a plugin named the same?

**Comment by tokejepsen** [comment]
BTW you don't need the ```@pyblish.api.log``` decorator anymore:)

**Comment by BigRoy** [comment]
> Do you mean it would conflict with other hosts default plugins?

It could conflict there as well. Basically anywhere where someone uses the same class name as these default plug-ins.

> Surely people would just use the data collected by those plugins, or they would need the plugin hence not having a plugin named the same?

Not entirely sure what you mean there. But yes, if they need additional plug-ins they could name it so it doesn't conflict. But I think plug-ins for different hosts with the same class name would still conflict since it seems based on class name (didn't test with different `hosts` attribute).

> BTW you don't need the @pyblish.api.log decorator anymore:)

Good point. Will remove that in #51 as well.

**Comment by tokejepsen** [comment]
> It could conflict there as well. Basically anywhere where someone uses the same class name as these default plug-ins.

Haven't experienced a problem with different hosts, but not on the latest pyblish either. Should still be the same though.

> For example CollectMayaCurrentFile and CollectMayaWorkspace.

Think you should just go ahead and renamed them any ways. Can't hurt:)

**Comment by mottosso** [comment]
Yep, I think this is a good initiative. Let's adopt it for other integrations as well.

Should the label also reflect that it's Maya-only?

**Comment by BigRoy** [comment]
> Should the label also reflect that it's Maya-only?

I think I already did that in the label that you just merged from #51. Sorry. ;) or did you mean more specific than that?

Anyway, will adapt class names and PR so we can close this up in a bit.

**Comment by mottosso** [comment]
> I think I already did that in the label that you just merged from #51. Sorry. ;) or did you mean more specific than that?

Yeah, I meant more as a question of *should* we do that. Let's go for that, merged and fixed!
